### ISSUE TITLE
Restriction lint for `/*` comments

### ISSUE THREAD

**Issue by joshtriplett** [rust-lang/rust-clippy#9784]
### What it does

This allow-by-default restriction lint detects any use of `/*` comments instead of `//` comments. This helps with codebases trying to maintain a uniform comment style, as well as codebases wanting to avoid the possibility of mid-line comments (with further code to the right of them).

Note: while in simple cases it *might* be easy to provide a rustfix suggestion, this lint has value even if it *doesn't* offer a suggestion. In complex cases I can easily imagine that providing a suggestion would require non-trivial changes.

### Lint Name

slash_star_comment

### Category

restriction

### Advantage

- Helps enforce uniform comment style.
- Avoids mid-line comments (`code /* comment */ more code`).
- Reduces visual noise

### Drawbacks

This is an intentionally opinionated restriction lint; it should remain allow-by-default, and should only be enabled by users wanting to check for it.

### Example

```rust
code(); /* comment */
```

Could be written as:

```rust
code(); // comment
```
